winter/pagination/page_position_argument_resolver.py

```diff
@@ -1,34 +1,82 @@
 import typing
+import warnings
 
+from furl import furl
 from rest_framework import exceptions
-from rest_framework.request import Request as DRFRequest
+from rest_framework.request import Request
 
+from .limits import LimitsAnnotation
+from .limits import MaximumLimitValueExceeded
 from .page_position import PagePosition
 from ..argument_resolver import ArgumentResolver
 from ..core import ComponentMethodArgument
+from ..exceptions import RedirectException
 
 
 class PagePositionArgumentResolver(ArgumentResolver):
 
-    def __init__(self, limit_parameter_name: str = 'limit', offset_parameter_name: str = 'offset'):
-        super().__init__()
-        self.limit_parameter_name = limit_parameter_name
-        self.offset_parameter_name = offset_parameter_name
+    def __init__(self, limit_name: str = 'limit', offset_name: str = 'offset'):
+        self.limit_name = limit_name
+        self.offset_name = offset_name
+        self.default_limit = None
+        self.maximum_limit = None
+        self.redirect_to_default_limit = False
 
     def is_supported(self, argument: ComponentMethodArgument) -> bool:
         return argument.type_ is PagePosition
 
-    def resolve_argument(self, argument: ComponentMethodArgument, http_request: DRFRequest) -> PagePosition:
-        raw_limit = http_request.query_params.get(self.limit_parameter_name)
-        raw_offset = http_request.query_params.get(self.offset_parameter_name)
+    def resolve_argument(self, argument: ComponentMethodArgument, http_request: Request) -> PagePosition:
+        raw_limit = http_request.query_params.get(self.limit_name)
+        raw_offset = http_request.query_params.get(self.offset_name)
 
         limit = self._parse_int_param(raw_limit, 'limit', min_value=1)
         offset = self._parse_int_param(raw_offset, 'offset', min_value=0)
+
+        limits_annotation = argument.method.annotations.get_one_or_none(LimitsAnnotation)
+
+        default_limit = self.default_limit
+        maximum_limit = self.maximum_limit
+        redirect_to_default_limit = self.redirect_to_default_limit
+
+        if limits_annotation is not None:
+            maximum_limit = limits_annotation.maximum
+            default_limit = limits_annotation.default
+            redirect_to_default_limit = limits_annotation.redirect_to_default
```

Comment:
Just remember your idea about `Limits` structure. It could allow extracting ~10 lines to a method like `_get_limits`:
```
page_position = self._parse_page_position(http_request)
limits = self._get_limits(argument)
page_position = self._apply_limits(page_position, limits)
return page_position
```

Reply:
done